File: 백준/Silver/18258. 큐 2/큐 2.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
        StringTokenizer st = new StringTokenizer(br.readLine());

        int n = Integer.parseInt(st.nextToken());
        Deque<Integer> queue = new LinkedList<>();
        for (int i = 0; i < n; i++) {
            st = new StringTokenizer(br.readLine());

            String cmd = st.nextToken();
            if (cmd.equals("push")) {
                queue.add(Integer.valueOf(st.nextToken()));
            } else if (cmd.equals("pop")) {
                if (queue.isEmpty()) {
                    bw.write("-1\n");
                } else {
                    bw.write(queue.poll() + "\n");
                }
            } else if (cmd.equals("size")) {
                bw.write(queue.size() + "\n");
            } else if (cmd.equals("empty")) {
                if (queue.isEmpty()) {
                    bw.write("1\n");
                } else {
                    bw.write("0\n");
                }
            } else if (cmd.equals("front")) {
                if (queue.isEmpty()) {
                    bw.write("-1\n");
                } else {
                    bw.write(queue.peek() + "\n");
                }
            } else {
                if (queue.isEmpty()) {
                    bw.write("-1\n");
                } else {
                    bw.write(queue.getLast() + "\n");
                }
            }

        }

        bw.flush();
    }
}

File: 백준/Silver/18258. 큐 2/큐 2.cc
#include <bits/stdc++.h>
using namespace std;

int main(void) {
	ios::sync_with_stdio(0);
	cin.tie(0);

	int n;
	cin >> n;
	
	queue<int> Q;
	while (n--) {
		string cmd;
		cin >> cmd;
		
		if (cmd == "push") {
			int x;
			cin >> x;
			Q.push(x);
		} else if (cmd == "pop") {
			if (Q.empty()) {
				cout << "-1\n";
			} else {
				cout << Q.front() << '\n';
				Q.pop();
			}
		} else if (cmd == "size") {
			cout << Q.size() << '\n';
		} else if (cmd == "empty") {
			if (Q.empty()) {
				cout << "1\n";
			} else {
				cout << "0\n";
			}
		} else if (cmd == "front") {
			if (Q.empty()) {
				cout << "-1\n";
			} else {
				cout << Q.front() << '\n';
			}
		} else { // cmd =="back"
			if (Q.empty()) {
				cout << "-1\n";
			} else {
				cout << Q.back() << '\n';
			}
		}
	}
	
	return 0;
}